akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/ResponseParsing.scala

```diff
@@ -66,18 +66,24 @@ private[http2] object ResponseParsing {
         }
 
       } else remainingHeaders.head match {
-        case (":status", statusCodeValue) =>
+        case (":status", statusCodeValue: String) =>
```

Comment:
Would make sense to convert this client-side match to a similar structure as on the server side. Let's postpone that to another PR.